[CI] MixedClusterEsqlSpecIT class failing

**Build Scans:**
- [elasticsearch-periodic #8298 / 8.12.2_bwc](https://gradle-enterprise.elastic.co/s/swpgbqcdnjvi6)
- [elasticsearch-periodic #8286 / 8.11.4_bwc](https://gradle-enterprise.elastic.co/s/tczz7ffzva3ci)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v8.12.2#javaRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT" -Dtests.method="test {string.MvStringNotEqualsFound SYNC}" -Dtests.seed=292C6160ADB10ADD -Dtests.bwc=true -Dtests.locale=mua-Latn-CM -Dtests.timezone=Africa/Lagos -Druntime.java=24
```

**Applicable branches:**
8.19

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Expected more data but no more entries found after [0]
Data mismatch:

Actual:
@timestamp:datetime | message:text

Expected:
@timestamp:datetime      | message:text                                                                                            
2023-10-23T13:55:01.546Z | More than one hundred characters long so it isn't indexed by the sub keyword field with ignore_above:100

```

**Issue Reasons:**
- [8.19] 2 failures in class org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT (1.8% fail rate in 113 executions)
- [8.19] 2 failures in pipeline elasticsearch-periodic (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/string.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/string.csv-spec
@@ -2304,6 +2304,8 @@ foo ( bar
 ;
 
 mvStringEquals
+required_capability: mv_warn
+
 FROM mv_text
 | WHERE message == "Connected to 10.1.0.1"
 | KEEP @timestamp, message
@@ -2316,6 +2318,8 @@ warning:Line 2:9: java.lang.IllegalArgumentException: single-value function enco
 ;
 
 mvStringEqualsLongString
+required_capability: mv_warn
+
 FROM mv_text
 | WHERE message == "More than one hundred characters long so it isn't indexed by the sub keyword field with ignore_above:100"
 | KEEP @timestamp, message
@@ -2328,6 +2332,8 @@ warning:Line 2:9: java.lang.IllegalArgumentException: single-value function enco
 ;
 
 mvStringNotEquals
+required_capability: mv_warn
+
 FROM mv_text
 | WHERE message != "Connected to 10.1.0.2"
 | KEEP @timestamp, message
@@ -2342,6 +2348,8 @@ warning:Line 2:9: java.lang.IllegalArgumentException: single-value function enco
 ;
 
 mvStringNotEqualsFound
+required_capability: mv_warn
+
 FROM mv_text
 | WHERE message != "Connected to 10.1.0.1"
 | KEEP @timestamp, message
@@ -2354,6 +2362,8 @@ warning:Line 2:9: java.lang.IllegalArgumentException: single-value function enco
 ;
 
 mvStringNotEqualsLong
+required_capability: mv_warn
+
 FROM mv_text
 | WHERE message != "More than one hundred characters long so it isn't indexed by the sub keyword field with ignore_above:100"
 | KEEP @timestamp, message

